### PR TITLE
broadcast free arp when pod is setup

### PR DESCRIPTION
**Pull request by changluyi**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
Examples of user facing changes:
- Features
- Bug fixes
- Docs
- Tests

### Which issue(s) this PR fixes:
Fixes #(issue-number)

### <samp>🤖 Generated by Copilot at 4e8d30a</samp>

This pull request adds a new feature to enable or disable IP conflict detection in the kube-ovn controller, and improves the ARP handling for pod IP addresses. It introduces a new flag `detectIPConflict` and a new utility function `BroadcastFreeArp` in the `util` package, and modifies the `configureNic` and `announce` functions in the `ovs_linux.go` and `ipam.go` files respectively.

### <samp>🤖 Generated by Copilot at 4e8d30a</samp>

> _`BroadcastFreeArp`_
> _A new function for pods_
> _To avoid conflict_

### <samp>🤖 Generated by Copilot at 4e8d30a</samp>

*  Add a new utility function to broadcast free ARP requests ([link](https://github.com/kubeovn/kube-ovn/pull/2638/files?diff=unified&w=0#diff-1fde66a27588df521a2a0fec2cfc0d6b59944eaab04e1e3fc31237c5e14fd246R224-R264))
* Use the new utility function to announce pod IP and MAC address after configuring the network interface ([link](https://github.com/kubeovn/kube-ovn/pull/2638/files?diff=unified&w=0#diff-9ef24f4ecb94bef2f6117b5b04dd931cc4cec65ab035472384bce083898827feR846-R851))

### CODE REVIEW

File: pkg/util/arp.go [hunk 1]
```diff
@@ -199,17 +199,45 @@ func ArpDetectIPConflict(nic, ip string, mac net.HardwareAddr) (net.HardwareAddr
 	// Announcement is identical to the ARP Probe described above,
 	// except that now the sender and target IP addresses are both
 	// set to the host's newly selected IPv4 address.
-	if pkt, err = arp.NewPacket(arp.OperationRequest, mac, tpa, tha, tpa); err != nil {
+	err = BroadcastFreeArp(nic, ip, mac, announceNum, announceInterval)
```
Comment (oilbeater) [inline pkg/util/arp.go:202]: if err = ...; err != nil

File: pkg/util/arp.go [hunk 2]
```diff
@@ -199,17 +199,45 @@ func ArpDetectIPConflict(nic, ip string, mac net.HardwareAddr) (net.HardwareAddr
...
+	if err != nil {
 		return nil, err
 	}
 
+	return nil, nil
+}
+
+func BroadcastFreeArp(nic, ip string, mac net.HardwareAddr, announceNum int, announceInterval time.Duration) error {
+	klog.Infof("broadcast free arp with nic %s , ip %s, with mac %v ", nic, ip, mac)
+	ifi, err := net.InterfaceByName(nic)
```
Comment (oilbeater) [inline pkg/util/arp.go:212]: ifi should be replaced by more meaningful name

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/util/arp.go (hunk 2 above)
Comment (changluyi) [reply pkg/util/arp.go:211]: replaced

File: pkg/util/arp.go (hunk 1 above)
Comment (changluyi) [reply pkg/util/arp.go]: fixed

File: pkg/util/arp.go (hunk 2 above)
Comment (changluyi) [reply pkg/util/arp.go]: fixed

**Review by oilbeater — APPROVED** [review]

**Pull request merged**